src/metabase/driver.clj

```diff
@@ -330,6 +330,14 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti execute-query-callback
+  "Like execute-query, but pass the result to a callback rather than returning it."
+  {:arglists '([driver query callback]), :style/indent 1}
```

Comment:
what happens if we run into an Exception when executing the query?

Reply:
For both the `sql-jdbc` and `:default` implementations, an exception is thrown as usual, and the callback is never invoked.